xbmc/video/VideoSelectActionProcessor.cpp

```diff
@@ -0,0 +1,152 @@
+/*
+ *  Copyright (C) 2023 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "VideoSelectActionProcessor.h"
+
+#include "FileItem.h"
+#include "ServiceBroker.h"
+#include "dialogs/GUIDialogContextMenu.h"
+#include "dialogs/GUIDialogSelect.h"
```

Comment:
It would be great if somehow this could be further decoupled. I mean isn't it possible to only have methods in this class which return a list of items and their respective action?
Opening the dialog select would fit better in the previous base class since it is in the scope of gui/guilib. A quick search on the phone seems to indicate all these gui includes are just required to open the dialog select. All the remaining code is pretty much gui free

Reply:
This class is intended to be a GUI helper. If every calling code would need to implement the dialog logic itself (in exactly the same way - as a code copy, as you want no helper ;-), this would contradict the idea of the refactoring a bit.

Reply:
Then can you please move the file itself to a guilib folder? e.g. video/guilib/XXX.cpp?
Would make the intention clear

Reply:
Or guilib/video/XXX.cpp?
It will be easy to find the "offending" consumers

Reply:
Okay, video/guilib/ it will be. I use this schema in pvr already...